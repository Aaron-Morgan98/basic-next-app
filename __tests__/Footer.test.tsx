import React from "react";
import { render } from "@testing-library/react";
import Footer from "../src/app/components/footer";

// Mocking the social media icons
jest.mock('@mui/icons-material/LinkedIn', () => () => <div data-testid="LinkedInIcon" />);
jest.mock('@mui/icons-material/GitHub', () => () => <div data-testid="GitHubIcon" />);

describe("Footer component", () => {
  it("renders correctly and matches snapshot", () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays the current year and author name", () => {
    const { getByText } = render(<Footer />);
    const currentYear = new Date().getFullYear();
    expect(getByText(`${currentYear}  Aaron Morgan`)).toBeInTheDocument();
  });

  it("renders LinkedIn and GitHub icons", () => {
    const { getByTestId } = render(<Footer />);
    // Assert thaticons are rendered
    expect(getByTestId("LinkedInIcon")).toBeInTheDocument();
    expect(getByTestId("GitHubIcon")).toBeInTheDocument();
  });
});
