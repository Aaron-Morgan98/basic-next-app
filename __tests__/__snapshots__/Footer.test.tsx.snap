// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="footerContainer"
  >
    <div
      class="footer"
    >
      <p>
        2024  Aaron Morgan
      </p>
    </div>
    <div
      class="icons"
    >
      <div
        data-testid="LinkedInIcon"
      />
      <div
        data-testid="GitHubIcon"
      />
    </div>
  </div>
</DocumentFragment>
`;
